src/main/java/at/hannibal2/skyhanni/test/TestCopyChestData.kt

```diff
@@ -0,0 +1,80 @@
+@file:Suppress("SENSELESS_COMPARISON") // intellij does a little trolling and assumes all ItemStacks will never be null. this suppresses such warnings.
+package at.hannibal2.skyhanni.test
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.test.command.CopyItemCommand.grabItemData
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.KeyboardManager.isKeyHeld
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.OSUtils
+import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraft.client.gui.inventory.GuiContainer
+import net.minecraft.inventory.Slot
+import net.minecraft.item.ItemStack
+import net.minecraftforge.client.event.GuiScreenEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+object TestCopyChestData {
+    private val config get() = SkyHanniMod.feature.dev.debug
+    private val mc = Minecraft.getMinecraft()
+    @SubscribeEvent
+    fun onKeybind(event: GuiScreenEvent.KeyboardInputEvent.Post) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (!config.copyPlayerInventory.isKeyHeld() && !config.copyEntireChest.isKeyHeld() && !config.copyChestName.isKeyHeld()) return
+        if (event.gui !is GuiContainer) return
+        if (config.copyEntireChest.isKeyHeld() && event.gui is GuiChest) {
+            copySlotsData(chest = (event.gui as GuiChest).inventorySlots.inventorySlots)
+            return
+        } else if (config.copyPlayerInventory.isKeyHeld()) {
+            copySlotsData(inventory = if (config.includeArmor) mc.thePlayer.inventory.mainInventory + mc.thePlayer.inventory.armorInventory else mc.thePlayer.inventory.mainInventory)
+            return
+        } else if (InventoryUtils.openInventoryName().isNotEmpty()) {
+            OSUtils.copyToClipboard(InventoryUtils.openInventoryName())
+            LorenzUtils.chat("Chest name copied to clipboard.")
+            return
+        }
+    }
+    private fun copySlotsData(chest: List<Slot> = emptyList(), inventory: Array<ItemStack> = emptyArray()) {
```

Comment:
I don't like this method having two disjoint behaviours depending on the arguments. Maybe split this method up into more methods (like one for the header, one for just listing item stacks, and one each for listing the inventory and one for the chests)

Reply:
split into two funcs, one test run and then commit